libd/src/d/semantic/expression.d

```diff
@@ -765,7 +844,13 @@ public:
 		import std.algorithm, std.array;
 		return callCallable(location, chooseOverload(location, s.set.map!((s) {
 			if (auto f = cast(Function) s) {
-				return getFrom(location, f);
+				auto c = getFrom(location, f);
+				if (c.type.kind != TypeKind.Error) {
+					return c;
+				} else {
+					//try to call as regular Function
+					return build!FunctionExpression(f.location, f);
+				}
```

Comment:
opEquals is a function where the "hasThis" flag is set to true.
But there is no way getThis can ever get this.
therefore it will error out saying it cannot find a suitable this pointer,
If that happens we build a FunctionExpression ourselves without going trough getFrom.